langchain4j-neo4j/src/main/java/dev/langchain4j/store/embedding/neo4j/Neo4jEmbeddingStore.java

```diff
@@ -181,60 +214,93 @@ private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<T
+                    this.sanitizedIdProperty
+            );
+            session.run(query);
+        }
     }
 
     private boolean indexExists() {
-        try (var session = session()) {
-            Map<String, Object> params = Map.of("name", this.indexName);
-            return session.run("SHOW INDEX WHERE type = 'VECTOR' AND name = $name", params)
-                    .hasNext();
+        try (Session session = session()) {
+            Map<String, Object> params = new HashMap<>();
+            params.put("name", this.indexName);
+            Result run = session.run("SHOW INDEX WHERE type = 'VECTOR' AND name = $name", params);
+            if (!run.hasNext()) {
+                return false;
+            }
+            Record record = run.single();
+            List<String> idxLabels = record
+                    .get("labelsOrTypes")
+                    .asList(Value::asString);
+            List<Object> idxProps = record.get("properties").asList();
+            
+            boolean isIndexDifferent = !idxLabels.equals(singletonList(this.label)) 
+                                       || !idxProps.equals(singletonList(this.embeddingProperty));
+            if (isIndexDifferent) {
+                String errMessage = String.format("It's not possible to create an index for the label `%s` and the property `%s`, \n" +
+                                 "as there is another index with name `%s` with different labels: `%s` and properties `%s`.\n" +
+                                 "Please provide another indexName to create the vector index, or delete the existing one",
+                        this.label,
+                        this.embeddingProperty,
+                        this.indexName,
+                        idxLabels,
+                        idxProps);
+                throw new RuntimeException(errMessage);
+            }
+            return true;
         }
     }
 
     private void createIndex() {
-        Map<String, Object> params = Map.of("indexName", this.indexName,
-                "label", this.label,
-                "embeddingProperty", this.embeddingProperty,
-                "dimension", this.dimension,
-                "distanceType", this.distanceType.getValue());
+        Map<String, Object> params = new HashMap<>();
+        params.put("indexName", this.indexName);
+        params.put("label", this.label);
+        params.put("embeddingProperty", this.embeddingProperty);
+        params.put("dimension", this.dimension);
+        params.put("distanceType", this.distanceType.getValue());
 
         // create vector index
-        try (var session = session()) {
+        try (Session session = session()) {
             session.run("CALL db.index.vector.createNodeIndex($indexName, $label, $embeddingProperty, $dimension, $distanceType)",
                     params);
 
-            session.run("CALL db.awaitIndexes()").consume();
+            session.run("CALL db.awaitIndexes(60)").consume();
```

Comment:
Ha detto Michael che il valore di default è troppo alto , e che, forse, è meglio abbassarlo
`https://github.com/langchain4j/langchain4j/pull/282#discussion_r1403332956`
```
the default is 300s (5min) here,
not sure if we want to pass in an explicit value that is large
enough for the index population to finish in case there is already data in the db.
```